packages/core/docs/commercetools/composables.md

```diff
@@ -31,32 +31,18 @@ const attributes = computed(() => productGetters.getAttributes(product.value[0])
 ::: slot example-product-search-custom-query
 ## Custom Query 
 
-There's an option to use your own Graphql query and variables while searching.   
+There's an option to use your own Graphql query and additional variables while searching by provising a second parameter to the composable. This parameter should be a function returning two properties:
+
+- `query` - GraphQL query that will be used instead of the default one;
+- `variables` - data object that will be shallowly merged (only one level deep, where provided keys have precedence) with data object provided as a first parameter.
 
 ```js
 import { useProduct } from '@vue-storefront/commercetools'
 
 const { search } = useProduct()
 
-search({ id: '12345' }, (query, variables) => ({ query, variables }))
-```
-
-At the same time you can have access to the default query or variables and overwrite them. 
-
-```js
-const customQuery = (query, variables) => {
-  // import a custom Graphql query
-  const newQuery = require('./queries/customQuery.gql')
-  // override "locale" variable and leave other ones untouched
-  const newVariables = { ...variabes, locale: 'en' }
-  
-  return {
-    query: newQuery,
-    variables: newVariables
-  }
-}
-search({ id: '12345' }, customQuery)
+search({ id: '12345' }, () => ({ query, variables }))
```

Comment:
Why you changed that doc? It's important to show that `variables` and `query` are accessible through fn params

Reply:
There are no params passed to this function. This is an old approach and docs weren't updated. This was confirmed with @lukasborawski